fmt/xm.c

```diff
@@ -576,6 +613,38 @@ static int load_xm_instruments(song_t *song, struct xm_file_header *hdr, slurp_t
 			ins->flags |= ENV_VOLUME | ENV_VOLSUSTAIN;
 		}
 
+		// panning loop endpoint fix, similar to volume fix above
+		if ((ins->flags & ENV_PANNING) && (ins->flags & ENV_PANLOOP)) {
+			l_start = ins->pan_env.loop_start;
+			l_end = ins->pan_env.loop_end;
+			if (l_start == l_end) {
+				ins->flags &= ~ENV_PANLOOP;
+			} else {
```

Comment:
Correct me if I'm wrong, but this whole block of code seems isomorphic with the corresponding block for volume envelopes above. It would save some LOC and some potential debugging if this block were "factored out" into a function that could be used for both.

Reply:
Yes, I wrote it out this way to make the patch changes more "readable" in a redundant way. I'll be happy to create a regular function to reduce future bugs and bring down the line count.